turbinia/workers/analysis/windows_acct.py

```diff
@@ -0,0 +1,142 @@
+    Returns:
+        TurbiniaTaskResult object.
+    """
+    # Where to store the resulting output file.
+    output_file_name = 'windows_account_analysis.txt'
+    output_file_path = os.path.join(self.output_dir, output_file_name)
+
+    # What type of evidence we should output.
+    output_evidence = ReportText(source_path=output_file_path)
+
+    try:
+      (location, num_files) = self._collect_windows_files(evidence)
+    except TurbiniaException as e:
+      result.close(
+          self, success=True,
+          status='No Windows account files found: {0:s}'.format(str(e)))
+      return result
+    if num_files < 2:
+      result.close(self, success=True, status='No Windows account files found')
+      return result
+    (creds, hashnames) = self._extract_windows_hashes(
+        result, os.path.join(location, 'Windows', 'System32', 'config'))
+    (report, priority, summary) = self._analyse_windows_creds(creds, hashnames)
+    output_evidence.text_data = report
+    result.report_priority = priority
+    result.report_data = report
+
+    # Write the report to the output file.
+    with open(output_file_path, 'wb') as fh:
+      fh.write(output_evidence.text_data.encode('utf-8'))
+
+    # Add the resulting evidence to the result object.
+    result.add_evidence(output_evidence, evidence.config)
+    result.close(self, success=True, status=summary)
+    return result
+
+  def _collect_windows_files(self, evidence):
+    try:
+      collected_artifacts = extract_artifacts(
+          artifact_names=['WindowsSystemRegistryFiles'],
+          disk_path=evidence.local_path, output_dir=self.output_dir)
+    except TurbiniaException as e:
+      raise TurbiniaException('artifact extraction failed: {}'.format(str(e)))
+
+    # Extract base dir from our list of collected artifacts
+    location = os.path.dirname(collected_artifacts[0])
+
+    return (location, len(collected_artifacts))
+
+  def _extract_windows_hashes(self, result, location):
+    # Dump the secrets into a file
+
+    # Default (empty) hash
+    IGNORE_CREDS = ['31d6cfe0d16ae931b73c59d7e0c089c0']
+
+    hash_file = os.path.join(self.tmp_dir, 'windows_hashes')
+    cmd = [
+        'secretsdump.py', '-system', location + '/SYSTEM', '-sam',
+        location + '/SAM', '-hashes', 'lmhash:nthash', 'LOCAL', '-outputfile',
+        hash_file
+    ]
+
+    impacket_log = os.path.join(self.output_dir, 'impacket.log')
+    self.execute(cmd, result, stdout_file=impacket_log)
+
+    creds = []
+    hashnames = {}
+    hash_file = hash_file + '.sam'
+    if os.path.isfile(hash_file):
+      with open(hash_file, 'r') as fh:
+        for line in fh:
+          (username, _, _, passwdhash, _, _, _) = line.split(':')
+          if passwdhash in IGNORE_CREDS:
+            continue
+          creds.append(line.strip())
+          hashnames[passwdhash] = username
+      os.remove(hash_file)
+    else:
+      result.close(self, success=False, status='Extracted hash file not found.')
```

Comment:
`result.close()` should only be called once when the task has completed (it also handles things like post-processing and uploading evidence, etc), and there is another call to this in the calling `run()` method which would also overwrite some of these values.  Maybe just add a log line with `result.log()` instead?  Though it would be nice to have this message in the status for the Task, so maybe either plumb back a status method, or maybe raising an exception here and closing out the result as unsuccessful similar to how `_collect_windows_files()` gets called?

Reply:
Done